<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Matthias Grenié &amp; Marten Winter" />


<title>01 - Context and Data Description</title>

<script src="site_libs/header-attrs-2.14/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/united.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>





<style type="text/css">
/* for pandoc --citeproc since 2.11 */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-inverse  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Biodiversity Facets Tutorial</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">
    <span class="fa fa-home"></span>
     
    Home
  </a>
</li>
<li>
  <a href="00-setup.html">
    <span class="fa fa-circle-info"></span>
     
    00-Setup
  </a>
</li>
<li>
  <a href="01-load-data.html">1 - Loading Data</a>
</li>
<li>
  <a href="02-functional-diversity.html">2 - Functional Diversity</a>
</li>
<li>
  <a href="03-phylogenetic-diversity.html">3 - Phylogenetic Diversity</a>
</li>
<li>
  <a href="04-stat-modelling.html">4 - Modelling</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">01 - Context and Data Description</h1>
<h4 class="author">Matthias Grenié &amp; Marten Winter</h4>
<h4 class="date">Monday April 29th 2022</h4>

</div>


<div id="context-of-the-study" class="section level1">
<h1>Context of the study</h1>
<p>We will be using the data from the following study <span
class="citation">(Döbert et al. 2017)</span>:</p>
<blockquote>
<p>Döbert, T.F., Webber, B.L., Sugau, J.B., Dickinson, K.J.M. and
Didham, R.K. (2017), Logging increases the functional and phylogenetic
dispersion of understorey plant communities in tropical lowland rain
forest. J Ecol, 105: 1235-1245. <a
href="https://doi.org/10.1111/1365-2745.12794"
class="uri">https://doi.org/10.1111/1365-2745.12794</a></p>
</blockquote>
<p>Logging is the major cause of forest degradation in the Tropics. The
effect of logging on taxonomic diversity is well known but more rarely
studied on other facets of biodiversity such as functional diversity and
phylogenetic diversity. Functional diversity and phylogenetic diversity
should better reflect the impact of logging on ecosystem. For example
logging can decrease the functional “redundancy” observed in ecosystems,
meaning that some functional traits could be lost.</p>
<p>The tropical lowland rain forests on the island of Borneo are
floristically among the most diverse systems on the planet, yet
large-scale timber extraction and conversion to commercial tree
plantations continue to drive their rapid degradation and loss. As is
the case for the majority of tropical forests, the effects of logging on
habitat quality in these forests have rarely been assessed, despite the
critical implications for biodiversity conservation. Moreover, studies
investigating the effects of logging on plant community dynamics across
both tropical and temperate forest ecosystems have rarely focused on the
understorey, despite its crucial relevance for successional
trajectories.</p>
<p>Our goal with this tutorial is to reproduce the analyses from the
paper and analyze how logging impacts the different facets of the
diversity of the understorey vegetation, and to reveal to what extent
similar facets give similar answers. The general goal is to familiarize
yourself with the data and functions needed to compute diversity
indices. As well the general principles behind them.</p>
</div>
<div id="loading-the-data" class="section level1">
<h1>Loading the data</h1>
<p>Fortunately for us, the authors of the study have shared openly the
data they used in their article <span class="citation">(Döbert et al.
2018)</span>. They are available through the Dryad platform at the
following link: <a href="https://doi.org/10.5061/dryad.f77p7"
class="uri">https://doi.org/10.5061/dryad.f77p7</a></p>
<blockquote>
<p>Döbert, Timm F. et al. (2018), Data from: Logging increases the
functional and phylogenetic dispersion of understorey plant communities
in tropical lowland rainforest, Dryad, Dataset, <a
href="https://doi.org/10.5061/dryad.f77p7"
class="uri">https://doi.org/10.5061/dryad.f77p7</a></p>
</blockquote>
<p>The fact that these data researchers provided the full dataset
including all data and meta-data will help us reproduce the exact same
analyses as well as additional analyses not in their paper.</p>
<div id="getting-the-data" class="section level2">
<h2>Getting the data</h2>
<p>To get the data you can follow the above-mentioned link <a
href="https://doi.org/10.5061/dryad.f77p7"
class="uri">https://doi.org/10.5061/dryad.f77p7</a> and click on the
“Download Dataset” button available on the top right of the webpage. It
will download a .zip file that you can unzip in the folder you created
for the project. This will create a folder named
<code>doi_10.5061_dryad.f77p7__v1</code> that contains all needed data
files.</p>
<p><strong>NOTE</strong>: We’ve also included a copy of the unzipped
file in the <code>data</code> folder of the practical project.</p>
</div>
<div id="summarizing-the-data" class="section level2">
<h2>Summarizing the data</h2>
<p>The zip file contains 5 files (also available in the
<code>data/doi_10.5061_dryad.f77p7__v1/</code> folder):</p>
<ul>
<li><code>README.txt</code> which is a text file that describes the
content of the other files with great precision. It details all the
columns available in the other files.</li>
<li><code>PlotData.csv</code> is a comma-separated file that describes
characteristics for each of the sampled vegetation plots including
logging metrics, environmental variables as well as taxonomic,
functional and phylogenetic diversity indices (to which we’ll compare
the indices we compute ourselves).</li>
<li><code>PlotSpeciesData.csv</code> is a comma-separated file that
contains a matrix of biomass values for the plant taxa sampled across
the sampled vegetation plots.</li>
<li><code>phylo_tree.nwk</code> the phylogenetic tree describing the
relationships between species.</li>
<li><code>SpeciesTraitData.csv</code> contains the complete list of
species sampled across all vegetation plots, with their associated
traits both continuous and discrete.</li>
</ul>
<p>We will load each of the file (apart from the phylogenetic tree) in
your workspace now with the <code>read.csv()</code> function:</p>
<pre class="r"><code>plot_data         = read.csv(&quot;data/doi_10.5061_dryad.f77p7__v1/PlotData.csv&quot;,
                             na.strings = c(&quot;NA&quot;, &quot;na&quot;),
                             stringsAsFactors = TRUE)
plot_species_data = read.csv(&quot;data/doi_10.5061_dryad.f77p7__v1/PlotSpeciesData.csv&quot;)
species_traits    = read.csv(&quot;data/doi_10.5061_dryad.f77p7__v1/SpeciesTraitData.csv&quot;,
                             na.strings = c(&quot;NA&quot;, &quot;na&quot;), stringsAsFactors = TRUE)</code></pre>
<p>To describe the data we will use the <code>str()</code>,
<code>summary()</code>, and <code>dim()</code> functions.</p>
<pre class="r"><code>str(plot_data)
summary(plot_data)

str(plot_species_data[, 1:5])
summary(head(plot_species_data)[,1:5])
dim(plot_species_data)

# Transform one column for further analyzed
species_traits$seed = ordered(species_traits$seed)
str(species_traits)
summary(species_traits)</code></pre>
<p><strong>Questions for you:</strong></p>
<ul>
<li><strong>Q1</strong>: How many plots were sampled?</li>
<li><strong>Q2</strong>: How many species are there in the dataset?</li>
<li><strong>Q3</strong>: How many traits are available?</li>
<li><strong>Q4</strong>: How many of them are continuous? How many of
them are discrete?</li>
<li><strong>Q5</strong>: What is the most numerous family among all
observed species?</li>
<li><strong>Q6</strong>: What is the most numerous genus?</li>
</ul>
</div>
<div id="description-of-the-environment-variables"
class="section level2">
<h2>Description of the environment variables</h2>
<p>Forest loss proportion is one of the main driver variable. The data
has been acquired across different block with different proportion of
logging and compared to unlogged forest.</p>
<pre class="r"><code>boxplot(forestloss17 ~ block, data = plot_data,
        xlab = &quot;Block of plot&quot;, ylab = &quot;Forest loss (%)&quot;,
        main = &quot;Forest loss in funciton of block of data&quot;)</code></pre>
</div>
</div>
<div id="references" class="section level1 unnumbered">
<h1 class="unnumbered">References</h1>
<div id="refs" class="references csl-bib-body hanging-indent">
<div id="ref-Dobert_Logging_2017" class="csl-entry">
Döbert, Timm F., Bruce L. Webber, John B. Sugau, Katharine J. M.
Dickinson, and Raphael K. Didham. 2017. <span>“Logging Increases the
Functional and Phylogenetic Dispersion of Understorey Plant Communities
in Tropical Lowland Rain Forest.”</span> <em>Journal of Ecology</em> 105
(5): 1235–45. <a
href="https://doi.org/10.1111/1365-2745.12794">https://doi.org/10.1111/1365-2745.12794</a>.
</div>
<div id="ref-Dobert_Data_2018" class="csl-entry">
———. 2018. <span>“Data from: <span>Logging</span> Increases the
Functional and Phylogenetic Dispersion of Understorey Plant Communities
in Tropical Lowland Rainforest.”</span> <span>Dryad</span>. <a
href="https://doi.org/10.5061/DRYAD.F77P7">https://doi.org/10.5061/DRYAD.F77P7</a>.
</div>
</div>
</div>



</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
